app/src/main/java/to/dev/dev_android/util/network/NetworkWatcher.kt

```diff
@@ -0,0 +1,27 @@
+package to.dev.dev_android.util.network
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.greenrobot.eventbus.EventBus
+import to.dev.dev_android.events.NetworkStatusEvent
+
+class NetworkWatcher(val coroutineScope: CoroutineScope) : BroadcastReceiver() {
+    companion object {
+        val intentFilter = IntentFilter().apply {
+            addAction("android.net.conn.CONNECTIVITY_CHANGE")
+        }
+    }
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        coroutineScope.launch {
+            if (async { NetworkUtils.isOnline() }.await()) {
+                EventBus.getDefault().post(NetworkStatusEvent(NetworkStatus.BACK_ONLINE))
```

Comment:
I added a breakpoint here and I wasn't able to see this line executed (meaning the `reload` didn't happen)

Reply:
The `reload` works fine. I tested it, and I can provide videos or gifs of it working on both Mobile network and on wifi. 

I just added a breakpoint on the `EventBus.post` line. I also wasn't able to see the line executed. But the event still got executed. This might be related to coroutines. I need to look more into it, because it was executed, but it didn't `break`.

Reply:
There was an exception being thrown. Due to `coroutineScope.launch` the exceptions get ignored. I solved an exception that was causing a malfunction on my device (I was only able to auto-refresh once before). 

If it still doesn't work for you. I need to research the issue. I'll need your device model and android version please.